tgui/packages/tgui/interfaces/PersonalCooking.tsx

```diff
@@ -0,0 +1,586 @@
+                        )}
+                      </Stack>
+                    </Tabs.Tab>
+                  ))}
+              </Tabs>
+              <hr style={{ 'border-color': '#111' }} />
+              <Button.Checkbox
+                fluid
+                content="Can make only"
+                checked={data.display_craftable_only}
+                onClick={() => {
+                  data.display_craftable_only = !data.display_craftable_only;
+                  act('toggle_recipes');
+                }}
+              />
+              <Button.Checkbox
+                fluid
+                content="Compact list"
+                checked={data.display_compact}
+                onClick={() => act('toggle_compact')}
+              />
+            </Section>
+          </Stack.Item>
+          <Stack.Item grow>
+            {recipes.length > 0 ? (
+              recipes
+                .slice(0, displayLimit)
+                .map((item) =>
+                  data.display_compact ? (
+                    <RecipeContentCompact
+                      key={item.ref}
+                      item={item}
+                      craftable={Boolean(craftabilityByRef[item.ref])}
+                      busy={data.busy}
+                    />
+                  ) : (
+                    <RecipeContent
+                      key={item.ref}
+                      item={item}
+                      diet={data.diet}
+                      craftable={Boolean(craftabilityByRef[item.ref])}
+                      busy={data.busy}
+                    />
+                  )
+                )
+            ) : (
+              <NoticeBox m={1} p={1}>
+                No recipes found.
+              </NoticeBox>
+            )}
+            {recipes.length > displayLimit && (
+              <Section textAlign="right">
+                And {recipes.length - displayLimit} more...
+              </Section>
+            )}
+          </Stack.Item>
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const TypeContent = (props) => {
+  return (
+    <Stack>
+      <Stack.Item width="14px" textAlign="center">
+        <Icon name={TYPE_ICONS[props.type]} />
+      </Stack.Item>
+      <Stack.Item grow style={{ 'text-transform': 'capitalize' }}>
+        {props.type.toLowerCase()}
+      </Stack.Item>
+      <Stack.Item>
+        {props.type === 'Can Make' ? (
+          props.craftable
+        ) : props.diet.liked_food.includes(props.type) ? (
+          <Icon name="face-laugh-beam" color={'good'} />
+        ) : props.diet.disliked_food.includes(props.type) ? (
+          <Icon name="face-tired" color={'average'} />
+        ) : (
+          props.diet.toxic_food.includes(props.type) && (
+            <Icon name="skull-crossbones" color={'bad'} />
+          )
+        )}
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+const RecipeContentCompact = (props, context) => {
+  const { act } = useBackend<Data>(context);
+  const item = props.item;
```

Comment:
```suggestion
  const { item } = props;
```
This is destructuring btw